packages/data/src/components/use-resolve-select/index.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * Internal dependencies
+ */
+import useSelect from '../use-select';
+
+/** @typedef {import('../../types').StoreDescriptor} StoreDescriptor */
+
+/**
+ * Like useSelect, but the selectors return objects containing
+ * both the original data AND the resolution info.
+ *
+ * @param {Function|StoreDescriptor|string} mapSelect see useSelect
+ * @param {Array}                           deps      see useSelect
+ * @return {Function} A custom react hook.
+ */
+export default function useResolveSelect( mapSelect, deps ) {
+	return useSelect( ( select, registry ) => {
+		// Memoization would be nice here
+		const resolve = ( store ) => enrichSelectors( select( store ) );
+		return mapSelect( resolve, select, registry );
```

Comment:
Is there a use case for the second `select` argument? `select.x()` is always equivalent to `resolve.x().data` so it's not strictly needed. I'm not sure about the meta-selectors like `getIsResolving`.

Reply:
I think you're right, let's remove it